yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1520,29 +1531,31 @@ cdef np.float64_t[:] rotation_matmul(np.float64_t[:, :] rotation_matrix,
             out[i] += rotation_matrix[i, j] * coordinate_matrix[j]
     return out
 
-
 @cython.boundscheck(False)
 @cython.wraparound(False)
-cpdef np.float64_t[:, :] get_rotation_matrix(normal_vector):
+cpdef np.float64_t[:, :] get_rotation_matrix(normal_vector, final_vector):
     """ Returns a numpy rotation matrix corresponding to the
-    rotation of the given normal vector to the z-axis ([0, 0, 1]).
+    rotation of the given normal vector to the specified final_vector.
     See https://math.stackexchange.com/a/476311 although note we return the
     inverse of what's specified there.
     """
     cdef np.float64_t[:] normal_vector_np = np.array([normal_vector[0], normal_vector[1], normal_vector[2]], 
                                                      dtype='float_')
-    cdef np.float64_t[:] z_axis = np.array([0., 0., 1.], dtype='float_')
+    cdef np.float64_t[:] final_vector_np = np.array([final_vector[0], final_vector[1], final_vector[2]], 
+                                                     dtype='float_')
     cdef np.float64_t[:] normal_unit_vector = normal_vector_np / np.linalg.norm(normal_vector_np)
-    cdef np.float64_t[:] v = np.cross(z_axis, normal_unit_vector)
+    cdef np.float64_t[:] final_unit_vector = final_vector_np / np.linalg.norm(final_vector_np)
+    cdef np.float64_t[:] v = np.cross(final_unit_vector, normal_unit_vector)
     cdef np.float64_t s = np.linalg.norm(v)
-    cdef np.float64_t c = np.dot(z_axis, normal_unit_vector)
-    # if the normal vector is identical to the z-axis, just return the
+    cdef np.float64_t c = np.dot(final_unit_vector, normal_unit_vector)
+    # if the normal vector is identical to the final vector, just return the
     # identity matrix
     if np.isclose(c, 1, rtol=1e-09):
         return np.identity(3, dtype='float_')
-    # if the normal vector is the negative z-axis, return:
+    # if the normal vector is the negative final vector, return the negative
+    # identity matrix:
     if np.isclose(s, 0, rtol=1e-09):
-        return np.array([[0, -1, 0],[-1, 0, 0],[0, 0, -1]], dtype='float_')
+        return -np.identity(3, dtype='float_')
```

Comment:
Why change the answer here?

Reply:
This method was originally defined to *only* calculate the rotation matrix for rotating to the z-axis.  I generalized it to work with any arbitrary vector (see the comment two lines up).  Thus, it seemed wrong to leave this manually calculated rotation matrix which was tied to a hard-coded z-axis.

Reply:
But the original answer was chosen because that's the limiting case when `s` goes to zero. Unless I'm misunderstanding I don't think you changed the math in a way that would cause the limiting case to go to the negative identity matrix instead of `[[0, -1, 0], [-1, 0, 0], [0, 0, -1]]`. Am I misunderstanding?

Reply:
In the old version of the code that was making a rotation matrix to rotate to the hard-coded z-axis, the conditional `np.isclose(s, 0, rtol=1e-09)` means something different.  If you just look at the old code, there are two hard-coded conditionals.  The first (line1542/1554) is looking at whether or not the input `normal_vector` is the positive z-axis.  If so, it sets the rotation matrix to be the identity matrix.  Since I'm removing the hard-coded z-axis from this function and generalizing it to rotate to any `final_vector`, I have changed this first check, to check to see if the `normal_vector` is the `final_vector` and then return the identity matrix.

In the second check, the one that your comment is regarding (line1544/1557) it was originally testing whether the `normal_vector` was the negative z-axis.  Again, I'm removing the hard-coded z-axis constraint and changing it to work with the `final_vector`. I'm not sure why the arbitrary rotation matrix was given `[[0, -1, 0], [-1, 0, 0], [0, 0, -1]]`.  But it seemed appropriate that for when the two vectors were anti-aligned to give the negative identity matrix, since we're giving the identity matrix when the two vectors are aligned and the same.  But maybe my linear algebra is wrong here.  

Do you know why the arbitrary rotation matrix was provided and why it should remain?  I honestly do not.

Reply:
Take a look at this example:

```
In [4]: np.array(get_rotation_matrix([0, 0, -1], [0, 0, 1]))
Out[4]:
array([[-1., -0., -0.],
       [-0., -1., -0.],
       [-0., -0., -1.]])

In [5]: np.array(get_rotation_matrix([1e-6, 1e-6, -1], [0, 0, 1]))
Out[5]:
array([[ 1.33126297e-04, -9.99866874e-01, -9.99467566e-07],
       [-9.99866874e-01,  1.33126297e-04, -9.99467566e-07],
       [ 9.99467566e-07,  9.99467566e-07, -9.99733747e-01]])
```

In the second case I'm choosing a `normal_vector` that is very close to `[0, 0, -1]` but not exactly equal. Note how your change is causing there to be a discontinuity in the return value of this function as we get close to `[0, 0, -1]`.

Reply:
OK, that's fair.  I didn't check close-by cases like this to figure out what the continuous behavior was.  I still don't know *why* things tend toward this arbitrary matrix (which I thought was tied to the z-axis hardcoding), but that's OK.  You've demonstrated that I should leave it as is.  This was the only part of my PR I wasn't 100% on, so thanks for sussing it out.  Code review for the win!